### PR TITLE
Decompile unk_0205FA2C.s

### PR DESCRIPTION
**Pull request by remicalixte**

### CODE REVIEW

**Review by red031000 — CHANGES_REQUESTED** [review]
if the spacing could be fixed I would be happy to merge this

**Comment by remicalixte** [comment]
Forgot to run `clang-format`

File: arm9/src/unk_0205FA2C.c [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+    0x00,
+    0x01,
+    0x02,
+    0x04,
+    0x03,
+    0x05,
+    0x06,
+    0x07,
+    0x08,
+    0x00,
```
Comment (PikalaxALT) [inline arm9/src/unk_0205FA2C.c:38]: This and the following two are implicit padding values for file boundary word alignment.

File: arm9/src/unk_0205FA2C.c [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+    }
+
+    FUN_0202A2C4(FUN_0202A8CC(sav2), 0xa, &res3);
+
+    return res3;
+}
+
+THUMB_FUNC BOOL FUN_02060144(u32 **param0)
+{
+    if (param0[7][0] == 0x1ED)
```
Comment (PikalaxALT) [inline arm9/src/unk_0205FA2C.c:540]: 0x1ED == 493 == SPECIES_ARCEUS

File: arm9/src/unk_0205FA2C.c (hunk 1 above)
Comment (remicalixte) [reply arm9/src/unk_0205FA2C.c:38]: Ah interesting. Is this always the case for constants at the top of a file?

Comment (PikalaxALT) [reply arm9/src/unk_0205FA2C.c]: It's the case when the object immediately following the array has some alignment forced upon it which is not satisfied by simply placing the new object at the end of the previous one.  For example, when a `u8[9]` is followed by a `u32`. The u32 needs to be 4-bytes aligned on ARM, so 0s are placed to fill the gap between the u8 array and the u32.

Comment (remicalixte) [reply arm9/src/unk_0205FA2C.c]: It makes sense, thanks. Here the array is followed by a function. I guess there are 4-bytes aligned too?

Comment (PikalaxALT) [reply arm9/src/unk_0205FA2C.c]: Not exactly. The order they appear in the C file is not the same as the order they appear in the final binary. Data arrays are placed in .rodata sections whereas functions are placed in .text sections.

**Comment by remicalixte** [comment]
Is there anything else I need to do before this PR can be merged?

**Review by red031000 — APPROVED** [review]

**Pull request merged**